chain/epoch_manager/src/reward_calculator.rs

```diff
@@ -268,6 +277,87 @@ mod tests {
         assert_eq!(result.1, 4_999_999u128);
     }
 
+    /// Test reward calculation for chunk only or block only producers
+    #[test]
+    fn test_reward_chunk_only_producer() {
+        let epoch_length = 1000;
+        let reward_calculator = RewardCalculator {
+            max_inflation_rate: Ratio::new(1, 100),
+            num_blocks_per_year: 1000,
+            epoch_length,
+            protocol_reward_rate: Ratio::new(0, 10),
+            protocol_treasury_account: "near".parse().unwrap(),
+            online_min_threshold: Ratio::new(9, 10),
+            online_max_threshold: Ratio::new(99, 100),
+            num_seconds_per_year: 1000,
+        };
+        let validator_block_chunk_stats = HashMap::from([
+            (
+                "test1".parse().unwrap(),
+                BlockChunkValidatorStats {
+                    block_stats: ValidatorStats { produced: 945, expected: 1000 },
+                    chunk_stats: ValidatorStats { produced: 945, expected: 1000 },
+                },
+            ),
+            // chunk only producer
+            (
+                "test2".parse().unwrap(),
+                BlockChunkValidatorStats {
+                    block_stats: ValidatorStats { produced: 0, expected: 0 },
+                    chunk_stats: ValidatorStats { produced: 999, expected: 1000 },
+                },
+            ),
+            // block only producer
+            (
+                "test3".parse().unwrap(),
+                BlockChunkValidatorStats {
+                    block_stats: ValidatorStats { produced: 945, expected: 1000 },
+                    chunk_stats: ValidatorStats { produced: 0, expected: 0 },
+                },
+            ),
+            // a validator that expected blocks and chunks are both 0
```

Comment:
Is such scenario possible right now ? (I'd mention it here in the comment)

Reply:
This could be possible if a validator with small stake is selected but it doesn't get assigned to produce any block (with some probability this will happen, although the probability is small).